website/src/auth.tsx

```diff
@@ -2,52 +2,71 @@ import { HubCallback } from "@aws-amplify/core"
 import Amplify, { Auth, Hub } from "aws-amplify"
 import React, { createContext, useContext, useState } from "react"
 
-const UserContext = createContext<CognitoUser | null>(null)
+import {
+  CognitoUser,
+} from 'amazon-cognito-identity-js'
+
+type UserContextType = {
+  user: CognitoUser | null;
+  setUser: (value: CognitoUser | null) => void;
+};
+
+export const UserContext = createContext<UserContextType>({} as UserContextType);
```

Comment:
General convention is not to export context objects directly (only `XProvider` and `useX`). So I'd say either all the code that needs to call `setUser` goes into one file, or exposing `setUser` via the hook (or a new hook). I lean toward the former because generally new code beyond this shouldn't need to call `setUser`.

Reply:
Done (I think) I made sure useContext() doesn't get called outside of the auth.tsx file, by creating new hooks. I also moved the login and logout functions to the UserProvider component.